### ISSUE TITLE
Link error with Android NDK

### ISSUE THREAD

**Issue by guanning79** [ispc/ispc#1313]
Hi,
I need to build a .so with ispc code for android mobile platform. And I used Android NDK r14b on Windows to cross compile the .so. 
I built ispc code on Windows using command like this: ispc.exe --woff some.ispc  -o some.ispc.o --arch=arm --target=neon-i32x4 -O3 -h some.ispc.h. And then linked some.ispc.o to .so.
In the .so there is a cpp code some.cpp to wrap ispc method to a dllexport method.
The compile command for some.cpp is:  
![default](https://user-images.githubusercontent.com/15059616/37966270-8f77d9d6-31fa-11e8-9f84-b5e02c7d3cb5.png)
The link command for .so is:
![default](https://user-images.githubusercontent.com/15059616/37966364-d043943c-31fa-11e8-9d86-540e1916dba5.png)
When linked .o which is compilation result of ispc code, this error message came out: uses VFP register arguments, output does not.
Does anyone has any idea about this, how can I make this cross-compilation things to work out?

**Comment by dbabokin** [comment]
Googling suggests that it has something with mixing hard and soft FP. I'm not familiar with ARM specifics, but if you identify what flags need to be passed to llvm back end, I may help doing that (it's done in ispc.cpp)

**Comment by guanning79** [comment]
I think the magic flag maybe "-mfloat-abi=softfp". 
In the beginning, I compiled dll .cpp file with this flag. And the error message "uses VFP register arguments, output does not" came out. I figured it's means ispc code was compiled with using VFP register but dll cpp was not. So I switch the flag to "-mfloat-abi=hard". But problem was not solved. Other link error came out, suach as: "android-ndk-r14b/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/libc++abi.a(cxa_exception.o) uses VFP register arguments, output does not", which made confused.

**Comment by guanning79** [comment]
Do you have any examples of using ispc on android?

**Comment by guanning79** [comment]
I have worked through this issue.
By using ispc to compile ispc code to llvm, and then using llc to convert llvm bitcode to asm. 
Because Android NDK use compilation flag: -mfloat-abi=softfp and .o which is compiled by ispc use flag: -mfloat-abi=hard, the error "uses VFP register arguments, output does not" will show up in link stage.
I had to compile ispc code to some universal format, llvm. Then convert llvm to asm by using llc and set flag -float-abi=soft.
The full llc command is: 
![default](https://user-images.githubusercontent.com/15059616/38015404-3b19195a-329e-11e8-8154-185765bb4a0e.png)
After this, compile cpp files and asm files, then link is successful.
